pkg/db/level/storage_trie_reader_test.go

```diff
@@ -2,49 +2,54 @@ package level_test
 
 import (
 	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/rlp"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	"github.com/vulcanize/eth-block-extractor/pkg/db/level"
 	"github.com/vulcanize/eth-block-extractor/test_helpers"
 	state_wrapper "github.com/vulcanize/eth-block-extractor/test_helpers/mocks/wrappers/core/state"
-	"github.com/vulcanize/eth-block-extractor/test_helpers/mocks/wrappers/rlp"
+	mock_rlp "github.com/vulcanize/eth-block-extractor/test_helpers/mocks/wrappers/rlp"
 	"github.com/vulcanize/eth-block-extractor/test_helpers/mocks/wrappers/trie"
 )
 
 var _ = Describe("Storage trie reader", func() {
 	It("decodes passed state trie leaf node into account", func() {
 		db := state_wrapper.NewMockStateDatabase()
 		trieDb := db.CreateFakeUnderlyingDatabase()
-		db.SetReturnDatabase(trieDb)
+		db.ReturnDB = trieDb
 		mockIteratror := trie.NewMockIterator(1)
 		mockIteratror.SetIncludeLeaf()
 		mockTrie := state_wrapper.NewMockTrie()
 		mockTrie.SetReturnIterator(mockIteratror)
-		db.SetReturnTrie(mockTrie)
-		decoder := rlp.NewMockDecoder()
-		decoder.SetReturnOut(&state.Account{})
+		db.ReturnTrie = mockTrie
+		decoder := mock_rlp.NewMockDecoder()
+		acct := new(state.Account)
+		err := rlp.DecodeBytes(test_helpers.FakeStateLeaf, acct)
```

Comment:
what's the rationale for deriving `acct` from `rlp.DecodeBytes` rather than setting the return out as `&state.Account{}`? Only asking because it requires the additional import of `go-ethereum/rlp` (in conjunction with `mocks/wrappers/rlp`

Reply:
I changed this because the [old test](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader_test.go#L26) fails with a 
```
  Expected error:
      <*errors.errorString | 0xc000213e10>: {s: "not found"}
      not found
  not to have occurred
```

When using the empty account and the test_helpers.FakeTrieNode, and from tracing this up this appears to be because there is an empty hash (0x0000000000000000000000000000000000000000000000000000000000000000) being passed to [trieDb.Node](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader.go#L40) in this case. 

This reminds me, I was a little confused by the [EmptyStorageTrieRoot](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader.go#L10), how was that value derived?

Reply:
I changed this because the [old test](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader_test.go#L26) fails with a 
```
  Expected error:
      <*errors.errorString | 0xc000213e10>: {s: "not found"}
      not found
  not to have occurred
```

When using the empty account and the test_helpers.FakeTrieNode, and from tracing this up this appears to be because there is an empty root hash (0x0000000000000000000000000000000000000000000000000000000000000000) being passed to [trieDb.Node](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader.go#L40) in this case. 

This reminds me, I was a little confused by the [EmptyStorageTrieRoot](https://github.com/vulcanize/eth-block-extractor/blob/master/pkg/db/level/storage_trie_reader.go#L10), how was that value derived?

Reply:
It's just a `[]byte` version of the empty root hash https://github.com/ethereum/go-ethereum/blob/master/core/state/statedb.go#L43

Reply:
Thank you! Can't yet give you a satisfactory answer as to why the test is failing that way with the empty `&state.Account{}`. This is part of the outdated code and won't be used for the syncAndPublish work, so maybe we ought to extract what we need for that into another repo and forget about trying to update all of this other stuff? Do you have a preference? My main concern with moving what we need to a new repo is preserving your git history.

Reply:
I'm not too worried about git history since that would be preserved here if you moved to a new repo. That said, I do really dig the idea of getting this repo up to date and working with the latest version of the deps. If what you're saying is that some of the proposed changes are out of scope for the story you're working on, I have no problem zeroing in on the task at hand and pushing those tasks back.

Reply:
No I think you are right, and this is pretty close to being up-to-date. Also I think I am on the cusp of getting ungx to work- will update shortly!